src/ripple/conditions/impl/PreimageSha256.h

```diff
@@ -0,0 +1,158 @@
+//------------------------------------------------------------------------------
+/*
+This file is part of rippled: https://github.com/ripple/rippled
+Copyright (c) 2016 Ripple Labs Inc.
+
+Permission to use, copy, modify, and/or distribute this software for any
+purpose  with  or without fee is hereby granted, provided that the above
+copyright notice and this permission notice appear in all copies.
+
+THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_CONDITIONS_PREIMAGE_SHA256_H
+#define RIPPLE_CONDITIONS_PREIMAGE_SHA256_H
+
+#include <ripple/basics/Buffer.h>
+#include <ripple/basics/Slice.h>
+#include <ripple/conditions/Condition.h>
+#include <ripple/conditions/Fulfillment.h>
+#include <ripple/conditions/impl/error.h>
+#include <ripple/protocol/digest.h>
+#include <memory>
+
+namespace ripple {
+namespace cryptoconditions {
+
+class PreimageSha256 final
+    : public Fulfillment
+{
+public:
+    /** The maximum allowed length of a preimage.
+
+        The specification does not specify a minimum supported
+        length, nor does it require all conditions to support
+        the same minimum length.
+
+        While future versions of this code will never lower
+        this limit, they may opt to raise it.
+    */
+    static constexpr std::size_t maxPreimageLength = 64;
+
+    /** Parse the payload for a PreimageSha256 condition
+
+        @param s A slice containing the DER encoded payload
+        @param ec indicates success or failure of the operation
+        @return the preimage, if successful; empty pointer otherwise.
+    */
+    static
+    std::unique_ptr<Fulfillment>
+    deserialize(
+        Slice s,
+        std::error_code& ec)
+    {
+        // Per the RFC, a preimage fulfulliment is defined as
+        // follows:
+        //
+        // PreimageFulfillment ::= SEQUENCE {
+        //     preimage             OCTET STRING
+        // }
+
+        using namespace der;
+
+        auto p = parsePreamble(s, ec);
+        if (ec)
+            return nullptr;
+
+        if (!isPrimitive(p))
+        {
+            ec = error::generic;
+            return {};
+        }
+
+        if (p.tag != 0)
```

Comment:
There is no max length, but we do impose a limit on the how large the conditions and fulfillments we accept are. See `maxPreimageLength`.